.github/workflows/python.yml

```diff
@@ -0,0 +1,120 @@
+# This file is autogenerated by maturin v1.4.0
+# To update, run
+#
+#    maturin generate-ci github -m pco_python/Cargo.toml
+#
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - master
+    tags:
+      - '*'
+  pull_request:
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
```

Comment:
If I understand right, none of these are actually running pytest; instead they're building and uploading candidate wheels. Can we change this to
1. only build/upload candidate wheels on release
2. pytest on a single platform, say the same Ubuntu I use in the Rust workflow?

We'll be deviating from the Maturin auto-generated config, but I think it's more of a suggestion than a target file.